Python/bltinmodule.c

```diff
@@ -1465,21 +1534,41 @@ map_reduce(mapobject *lz, PyObject *Py_UNUSED(ignored))
         PyTuple_SET_ITEM(args, i+1, Py_NewRef(it));
     }
 
+    if (lz->strict) {
+        return Py_BuildValue("ONO", Py_TYPE(lz), args, Py_True);
+    }
     return Py_BuildValue("ON", Py_TYPE(lz), args);
 }
 
+PyDoc_STRVAR(setstate_doc, "Set state information for unpickling.");
+
+static PyObject *
+map_setstate(mapobject *lz, PyObject *state)
+{
+    int strict = PyObject_IsTrue(state);
```

Comment:
Can this be a tuple in the future? Maybe compare to actual Py_True.

Reply:
This is also what `zip()` does, should I deviate?

Reply:
I'd prefer if you deviate, to make this a bit more forward-compatible/future-proof.